DuckDuckGo.Fluent.Plugin/AppFunctions.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Avalonia.Threading;
+using Blast.API.Core.Processes;
+using Blast.API.Graphics;
+using Blast.API.Processes;
+using Blast.Core.Results;
+using SoftCircuits.HtmlMonkey;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+using static DuckDuckGo.Fluent.Plugin.DuckDuckGoSearchResult;
+using static DuckDuckGo.Fluent.Plugin.DuckDuckGoSearchApp;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    public class AppFunctions
+    {
+        public enum ResultType
+        {
+            Answer,
+            Definition,
+            Abstract,
+            QrCode,
+            SearchResult
+        }
+
+        private const string Endpoint = "https://api.duckduckgo.com/?q=";
+        public const string DuckWebsiteUrl = "https://duckduckgo.com/";
+
+        public static string GetEndpointUrl(string searchedText)
+        {
+            return Endpoint + WebUtility.UrlEncode(searchedText) + "&format=json&no_html=1";
+        }
+
+        public static bool VerifySearchedTerms(ref string searchedText, ref string searchedTag)
+        {
+            searchedText = searchedText.Trim();
+
+            if (string.IsNullOrWhiteSpace(searchedText)) return false;
+
+            if (string.IsNullOrWhiteSpace(searchedTag)) return true;
+
+            return searchedTag.Equals(DuckSearchTagName, StringComparison.Ordinal) ||
+                   searchedTag.Equals(QrTag, StringComparison.Ordinal);
+        }
+
+        public static async Task GetRelatedSearch(Root root, Channel<DuckResult> channel, string searchedText)
+        {
+            if (root?.RelatedTopics != null)
+                foreach (RelatedTopic variableTopic in root.RelatedTopics.Where(variableTopic =>
+                    variableTopic != null))
+                {
+                    if (!string.IsNullOrWhiteSpace(variableTopic.Text))
+                        await channel.Writer.WriteAsync(new DuckResult
+                        {
+                            Info = variableTopic.Text, SourceUrl = variableTopic.FirstUrl, ResultType = "Related",
+                            Score = 3,
```

Comment:
Problem is: If I use `SearchDistance` to give score to `variableTopic` results, it may come on top of `GetAnswers` and `GetDictionary` results. Just like in Search Engines, I just wanted Instant Answers to be always on top (or pinned in other sense).